clippy_lints/src/minmax.rs

```diff
@@ -87,16 +87,17 @@ fn fetch_const<'a>(
     if args.len() != 2 {
         return None;
     }
-    if let Some(c) = constant_simple(cx, cx.tables, &args[0]) {
-        if constant_simple(cx, cx.tables, &args[1]).is_none() {
-            // otherwise ignore
-            Some((m, c, &args[1]))
-        } else {
-            None
+    constant_simple(cx, cx.tables, &args[0]).map_or_else(
```

Comment:
I'm ok with this change, but generally don't think this lint should trigger, if there is a `else if let` arm in addition to an `if let` arm.